flang/lib/Common/CMakeLists.txt

```diff
@@ -44,5 +44,6 @@ add_flang_library(FortranCommon
   ${version_inc}
 
   LINK_COMPONENTS
+  Passes
```

Comment:
Is this dependency really needed?

Reply:
Yes. The default optimization level is `llvm::OptimizationLevel::O0`. This is a class instance which is defined in `libLLVMPasses`. Without it, one gets an undefined reference error when `-DBUILD_SHARED_LIBS=ON`